structuralcodes/codes/mc2010/_concrete_shear.py

```diff
@@ -0,0 +1,1598 @@
+    rather then around, reducing the crack roughness
+
+    fib Model Code 2010, Eq. (7.3-17), (7.3-20) and (7.3-21)
+
+    Args:
+        fck (float): Characteristic strength in MPa
+        z (float): The length to the areasenter of cross-section in mm
+        bw (float): Thickness of web in cross section
+        dg (float): Maximum size of aggregate
+        E_s (float): The E_s-modulus to the materialb in MPa
+        As (float): The cross-section area in mm^2
+        Med (float): The moment working on the material in Nmm
+        Ved (float): The shear working on the material in N
+        Ned (float): The normal force working on the material in N
+        delta_e (float): The exentricity of the load in mm
+        gamma_c (float): Safety factor
+
+    Returns:
+        float: Design shear resistance without shear reinforcement
+    """
+    fsqr = min(fck**0.5, 8)
+
+    k_dg = max(32 / (16 + dg), 0.75)
+    kv = (0.4 / (1 + 1500 * epsilon_x(E_s, As, Med, Ved, Ned, z, delta_e))) * (
+        1300 / (1000 + k_dg * z)
+    )
+    return (kv * fsqr * z * bw) / gamma_c
+
+
+def v_rdc_approx3(  # tror dette egentlig er vrds3
+    approx_lvl_s: float,
+    fck: float,
+    z: float,
+    bw: float,
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    delta_e: float,
+    alfa: float,
+    gamma_c: float = 1.5,
+) -> float:
+    """The design shear resistance of a web or a slab without
+    shear reinforcement.
+
+    fib Model Code 2010, Eq. (7.3-17), (7.3-39) and (7.3-43)
+
+    Args:
+        fck (float): Characteristic strength in MPa
+        z: (float): The length to the areasenter of cross-section in mm
+        bw: (float): Thickness of web in cross section
+        dg: (float): Maximum size of aggregate
+        E_s: (float): The E_s-modulus to the materialb in MPa
+        As: (float): The cross-section area in mm^2
+        Med: (float): The moment working on the material in Nmm
+        Ved: (float): The shear working on the material in N
+        Ned: (float): The normal force working on the material in N
+        delta_e (float): The exentricity of the load in mm
+        alfa (float): Inclination of the stirrups in degrees
+        gamma_c (float): Safety factor
+
+    Returns:
+        float: Design shear resistance without shear reinforcement
+    """
+    fsqr = min(fck**0.5, 8)
+    theta_min = 20 + 10000 * epsilon_x(E_s, As, Med, Ved, Ned, z, delta_e)
+    kv = max(
+        (0.4 / (1 + 1500 * epsilon_x(E_s, As, Med, Ved, Ned, z, delta_e)))
+        * (
+            1
+            - Ved
+            / v_rd_max(
+                approx_lvl_s,
+                fck,
+                bw,
+                theta_min,
+                z,
+                E_s,
+                As,
+                Med,
+                Ved,
+                Ned,
+                delta_e,
+                alfa,
+                gamma_c,
+            )
```

Comment:
Calculate v_rd_max in a separate line, e.g. after `theta_min` and before `kv` to reduce the complexity of this expression.